src/stan/services/variational/print_progress.hpp

```diff
@@ -0,0 +1,41 @@
+#ifndef STAN_SERVICES_VARIATIONAL_PRINT_PROGRESS_HPP
+#define STAN_SERVICES_VARIATIONAL_PRINT_PROGRESS_HPP
+
+#include <stan/services/io/do_print.hpp>
+#include <cmath>
+#include <iomanip>
+#include <iostream>
+#include <string>
+
+namespace stan {
+  namespace services {
+    namespace variational {
+
+      void print_progress(const int m,
+                          const int start,
+                          const int finish,
+                          const int refresh,
+                          const bool tune,
+                          const std::string prefix,
+                          const std::string suffix,
+                          std::ostream& o) {
+        int it_print_width = std::ceil(std::log10(static_cast<double>(finish)));
```

Comment:
Add a unit test for this function. Since there's no error checking here and no doc, what are valid values of `start`, `finish`, `refresh`, `tune`, etc. that should be tested?